kaprekar_constant.cpp

```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include <algorithm>
+
+int getDescendingNumber(int num) {
+    std::string numStr = std::to_string(num);
+    std::sort(numStr.begin(), numStr.end(), std::greater<char>());
+    return std::stoi(numStr);
+}
+
+int getAscendingNumber(int num) {
+    std::string numStr = std::to_string(num);
+    std::sort(numStr.begin(), numStr.end());
+    return std::stoi(numStr);
+}
+
+void kaprekarRoutine(int num) {
+    while (num != 6174) {
+        int descNum = getDescendingNumber(num);
+        int ascNum = getAscendingNumber(num);
+        num = descNum - ascNum;
+        std::cout << descNum << " - " << ascNum << " = " << num << std::endl;
+    }
+}
+
+int main() {
+    int initialNum;

```

Comment:
variable name is not a clearing purpose of the variable. It should be userInput, inputNumber